### PR TITLE
Fix issues identified with static code analysis

### PR DESCRIPTION
**Pull request by gwenneg**
As stated in the title, I fixed some issues identified during static code analysis.

### CODE REVIEW

**Comment by gwenneg** [comment]
Please note that the package of `AmazonLambdaHttpProcessor` was fixed. It doesn’t look breaking to me, but I’d like a second opinion on that change.

**Comment by geoand** [comment]
Thanks for this!

Although the changes do look good and seem perfectly reasonable, perhaps they are bit too broad?

**Comment by gwenneg** [comment]
You're right, I'll narrow it down to the essential changes and leave the minor fixes aside.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ public Result parse(MethodInfo methodInfo) {
         boolean allIgnoreCase = false;
         if (afterByPart.contains(ALL_IGNORE_CASE)) {
             allIgnoreCase = true;
-            afterByPart = afterByPart.replaceAll(ALL_IGNORE_CASE, "");
+            afterByPart = afterByPart.replace(ALL_IGNORE_CASE, "");
```
Comment (gwenneg) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java]: This prevents a regex pattern compilation.

**Comment by gwenneg** [comment]
Done! This PR is mostly about unclosed InputStreams now.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java (hunk 1 above)
Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/MethodNameParser.java]: Interesting!

**Review by geoand — APPROVED** [review]
Thanks @gwenneg!

**Comment by gwenneg** [comment]
Thanks for reviewing this :)

**Pull request merged**